Marlin/Configuration_adv.h

```diff
@@ -1132,6 +1132,136 @@
+ *  commenting that function out (or not enabling it) and assigning it's pin number to the
+ *  speed pin.
+ *
+ *  For all CPUs the hardware PWMs on TIMER1 are not available.  Marlin uses TIMER1 to generate
+ *  interrupts and sets it up in such a way that the none of it's PWMs can be used.
+ *
+ *  Servos also make hardware PWM(s) unavailable.  In this case it's only the "A" PWM that's
+ *  unavailable.  The other hardware PWM(s) on that timer are available for general use.
+ *
+ *  Below is a table that can be used to when selecting the speed pin on a 2560.  Other CPUs
+ *  are a subset of the 2560.
+ *
+ *  ATmega2560 PWM assignments & users
+ *
+ *  There are 16 PWM ports assigned to 15 physical pins.
+ *     Pin 13 has two ports assigned to it.
+ *
+ *      Timer   Digital Normally    Used by         Optional
+ *      & Port   Pin    assigned    system          users
+ *      TIMER3B   2     X_MAX
+ *      TIMER3C   3     X_MIN
+ *      TIMER0B   4     HEATER_4
+ *      TIMER3A   5     HEATER_5                    *servo 0-11 ISR
+ *      TIMER4A   6     HEATER_6                    *servo 12-23 ISR
+ *      TIMER4B   7     LCD
+ *      TIMER4C   8     HOTBED
+ *      TIMER2B   9     HEATER_1
+ *      TIMER2A   10    HEATER_0
+ *      TIMER1A   11    HEATER_7    *stepper ISR
+ *      TIMER1B   12    PS_ON_PIN   *stepper ISR
+ *      TIMER0A   13    LED         LED PWM
+ *      TIMER1C   13                *stepper ISR
+ *      TIMER5C   44    LCD                         stepper motor current XY PWM
+ *      TIMER5B   45    LCD                         stepper motor current Z PWM
+ *      TIMER5A   46    Z_STEP                      stepper motor current E PWM or *servo 24-35 ISR
+ *
+ *   * - These hardware PWMs are not available.  The pin can still be used as a general purpose
+ *       digital I/O.
+ *
+ *  In addition to the above, fans can be assigned to PWM pins.  If you pick a pin that's
+ *  already assigned to a fan then you'll need to delete the fan or change its pin assignment.
+ *  This needs to be done even if FAN_FAST_PWM is disabled.
+ *
+ *  NOTE: Most pins that are hardwired to a heater or a fan usually are driven by a power driver
+ *        MOSFET which has on it's output a pull up through an LED to +12V/+24V. This will
+ *        probably damage your spindle controller unless you add some circuitry.  If there isn't
+ *        a +12V/+24V pull up you'll need an external 1k-10k pull up resistor to the pin.
+ *
+ *
+ *  AT90USB646, 647, 1286 & 1287 PWM assignments
+ *
+ *     As with the 2560, the PWMs on Timer1 are not available.
+ *
+ *     These chips have 10 PWMs assigned to 9 pins.  TIMER0A and TIMER1C are tied to the same pin.
+ *     Most Arduino IDE extensions only make TIMER1C available (Teensyduino included).
+ *
+ *
+ *  ATmega644 & 1284 PWM assignments
+ *
+ *     As with the 2560, the PWMs on Timer1 are not available.
+ *
+ *     All PWMs have their own pins.
+ *
+ */  //end SPINDLE LASER
```

Comment:
This is simply too much documentation to add to the config files. Reduce it to the minimum and use links to the relevant webpages where information can be found. If you need very detailed documentation, add it to the `MarlinDocumentation` repository, not the configs.